From 15d123009b7e987e946309cd5724ae17f376bd16 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Sun, 18 Dec 2022 16:26:02 +0100
Subject: [PATCH] GCC: fix selection of IMMEDIATE_CRASH

immediate_crash.h still uses V8_CC_GCC define to determine which
IMMEDIATE_CRASH macro is used. This should be V8_CC_GNU instead.
Otherwise weird compile errors are happening with turboshaft.

Bug: chromium:819294
Change-Id: Id77fe7406ae16a804e1e466844f81d6c728ec008
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4111849
Commit-Queue: Stephan Hartmann <stha09@googlemail.com>
Reviewed-by: Michael Lippautz <mlippautz@chromium.org>
Cr-Commit-Position: refs/heads/main@{#84923}
---

diff --git a/src/base/immediate-crash.h b/src/base/immediate-crash.h
index 770cb27..ce6240b 100644
--- a/v8/src/base/immediate-crash.h
+++ b/v8/src/base/immediate-crash.h
@@ -140,9 +140,9 @@
     [] { TRAP_SEQUENCE_(); }();  \
   } while (false)
 
-#endif  // !V8_CC_GCC
+#endif  // !V8_CC_GNU
 
-#if defined(__clang__) || V8_CC_GCC
+#if defined(__clang__) || V8_CC_GNU
 
 // __builtin_unreachable() hints to the compiler that this is noreturn and can
 // be packed in the function epilogue.
